burn-tensor/src/tensor/api/float.rs

```diff
@@ -72,6 +72,11 @@ where
         Self::new(B::sqrt(self.primitive))
     }
 
+    /// Applies element wise absolute value operation.
+    pub fn abs(self) -> Self {
+        Self::new(B::abs(self.primitive))
+    }
```

Comment:
You should use the `numeric` API instead, you can look at how `sum` is implemented there. https://github.com/burn-rs/burn/blob/f453f47efb1887ed736faf03157530931b2d3d18/burn-tensor/src/tensor/api/numeric.rs#L134C14-L134C14

Reply:
@nathanielsimard what about the backward pass implementation? Was your suggestion is for exact optimization?

Reply:
Yes, it's mostly to reduce the number of kernels executed when using `abs` with autodiff. The other review is still valid, this is just my new comments about the `int` and `float` implementations.

Reply:
I have committed changes for numeric API. 
However, I would be grateful of explanation, why it should be there

Reply:
If you want to work on documentation, maybe I could help you a little bit, by asking questions :D

Reply:
> I have committed changes for numeric API. However, I would be grateful for explanation, why it should be there

@nathanielsimard knows the full answer but as I understood numeric if for int and float at the same time. If you have a specific behavior only for float than it goes into float. 

@nathanielsimard at some we'll need more explanation from you otherwise I'll be telling people legends =P